### PR TITLE
ButtonGroup component

### PR DESCRIPTION
**Pull request by manuelpuyol**
This PR creates a `ButtonGroupComponent` based on https://primer.style/css/components/buttons#button-groups and https://primer.style/components/Buttons#default-examples
I decided to make the buttons slots to make it harder to use this component incorrectly, so users don't add things that shouldn't be inside the BtnGroup

image

### CODE REVIEW

File: app/components/primer/button_group_component.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use ButtonGroupComponent to wrap a series of buttons.
+  # They will be rounded and spaced automatically
```
Comment (joelhawksley) [inline app/components/primer/button_group_component.rb:5]:
```suggestion
  # Use ButtonGroupComponent to render a series of buttons.
```

File: app/components/primer/button_group_component.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  class ButtonGroupComponent < Primer::Component
+
```
Comment (joelhawksley) [inline app/components/primer/button_group_component.rb:7]:
```suggestion
```

File: stories/primer/button_group_component_stories.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+class Primer::ButtonGroupComponentStories < ViewComponent::Storybook::Stories
+  layout "storybook_preview"
+
+  story(:button_group) do
+    content do |component|
+      component.button { "Button" }
+      component.button { "Button" }
+      component.button { "Button" }
+      component.button { "Button" }
```
Comment (joelhawksley) [inline stories/primer/button_group_component_stories.rb:11]: Might we use the example from the component class here that has the different button types?

File: test/components/button_group_component_test.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
```
Comment (joelhawksley) [inline test/components/button_group_component_test.rb:1]: What should we do if no buttons are declared?

File: test/components/button_group_component_test.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  def test_renders_button_items
+    render_inline(Primer::ButtonGroupComponent.new) { |c| c.button { "Button" } }
+
+    assert_selector("div.BtnGroup") do
+      assert_selector("button.btn.BtnGroup-item", text: "Button")
+    end
+  end
+
+  def test_renders_button_with_props
```
Comment (joelhawksley) [inline test/components/button_group_component_test.rb:22]:
```suggestion
  def test_renders_button_with_arguments
```

File: test/components/button_group_component_test.rb (hunk 4 above)
Comment (manuelpuyol) [reply test/components/button_group_component_test.rb:1]: I guess not rendering is the best idea, will update

File: app/components/primer/button_group_component.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use ButtonGroupComponent to render a series of buttons.
+  class ButtonGroupComponent < Primer::Component
+    include ViewComponent::SlotableV2
+
+    renders_many :buttons, ->(classes: "", **kwargs) do
```
Comment (joelhawksley) [inline app/components/primer/button_group_component.rb:8]: What do you think is the best way to document that these slots take all of the arguments for ButtonComponent?

Comment (manuelpuyol) [reply app/components/primer/button_group_component.rb:8]: That's a good question. I don't think we aren't documenting slots in other components
@srt32 wdyt?

Comment (srt32) [reply app/components/primer/button_group_component.rb:8]: I've seen us document slots v1 behavior like this https://github.com/primer/view_components/pull/182/files#diff-a870f147518fb3db01d7aacb41bc479052f20a68f7e0f934fc78715d1851dd68R36-R39. We'd have to rewrite the Rakefile parser to handle slots v2?

Comment (srt32) [reply app/components/primer/button_group_component.rb:8]: Continued over in https://github.com/primer/view_components/pull/185#discussion_r569558545

**Pull request closed without merging**